synapseclient/core/download/download_async.py

```diff
@@ -326,169 +266,286 @@ class _MultithreadedDownloader:
     from a URL that supports range headers.
     """
 
-    def __init__(self, syn, executor, max_concurrent_parts):
+    def __init__(
+        self,
+        syn: "Synapse",
+        download_request: DownloadRequest,
+    ) -> None:
         """
         Initializes the class
 
         Arguments:
             syn: A synapseclient
             executor: An ExecutorService that will be used to run part downloads
                          in separate threads
-            max_concurrent_parts: An integer to specify the maximum number of
-                                    concurrent parts that can be downloaded at once.
-                                    If there are more parts than can be run concurrently
-                                    they will be scheduled in the executor when previously
-                                    running part downloads complete.
         """
         self._syn = syn
-        self._executor = executor
-        self._max_concurrent_parts = max_concurrent_parts
+        self._thread_lock = _threading.Lock()
+        self._aborted = False
+        self._download_request = download_request
+        self._progress_bar = None
+        self._should_close_progress_bar = (
+            getattr(_thread_local, "progress_bar", None) is None
+        )
 
-    def download_file(self, request: DownloadRequest) -> None:
+    async def download_file(self) -> None:
         """
         Splits up and downloads a file in chunks from a URL.
 
         Arguments:
             request: A DownloadRequest object specifying what Synapse file to download.
```

Comment:
Request is no longer an arg and should be removed

Reply:
Removed in the second PR
https://github.com/Sage-Bionetworks/synapsePythonClient/pull/1111